syntax = "proto3";

package crm;

service CustomerService{
    rpc CreateCustomer(CreateCustomerRequest) returns (CustomerResponse);
    rpc GetCustomer(GetCustomerRequest) returns (CustomerResponse);
    rpc UpdateCustomer(UpdateCustomerRequest) returns (CustomerResponse);
    rpc DeleteCustomer(DeleteCustomerRequest) returns (Empty);
    rpc ListCustomers(Empty) returns (CustomersListResponse);
}

message Customer{
    string id = 1;
    string name = 2;
    string email = 3;
    string created_at = 4;
}

message CreateCustomerRequest{
    string name = 1;
    string email = 2;
}

message GetCustomerRequest{
    string id = 1;
}

message UpdateCustomerRequest{
    string id = 1;
    string name = 2;
    string email = 3;
}

message DeleteCustomerRequest{
    string id = 1;
}

message Empty{}

message CustomerResponse{
    Customer customer = 1;
}

message CustomersListResponse{
    repeated Customer customers = 1;
}

service OrderService{
    rpc CreateOrder(CreateOrdersRequest) returns (OrderResponse);
    rpc GetCustomerOrders(GetCustomerOrdersRequest) returns (OrderListResponse);
}

message Order{
    string id = 1;
    string customer_id = 2;
    string product_name = 3;
    double price = 4;
    string created_at = 5;
}

message CreateOrdersRequest{
    string customer_id = 1;
    string product_name = 2;
    double price = 3;
}

message GetCustomerOrdersRequest{
    string customer_id = 1;
}

message OrderResponse{
    Order order = 1;
}

message OrderListResponse{
    repeated Order orders = 1;
}